### ISSUE TITLE
Separate data & UI modules

### ISSUE THREAD

**Issue by erawhctim** [ChuckerTeam/chucker#1002]
## :warning: Is your feature request related to a problem? Please describe
My request was revealed as I ran into the same problem as #801 (but ultimately not related to a problem, just an enhancement request)

## :bulb: Describe the solution you'd like
It'd be great if Chucker could isolate its UI into a standalone artifact, separate from the "core" Chucker dependency, so clients could optionally use Chucker's built-in UI or build their own. I would say this problem is _somewhat_ related to #801 and #579.

In its current form, the main chucker artifact requires consumers to be on the same versions of the Android/Jetpack libraries that the Chucker UI relies on (and pulling in those transitive dependencies, when not on the same versions, can cause unintended side effects). 

## :bar_chart: Describe alternatives you've considered
- Upgrading all my app's dependencies to match the versions that Chucker uses
- Trying to isolate the Chucker dependency into its own module to somehow circumvent the transitive dependency issue (I'm not smart enough in Gradle to figure this one out 😅)

## :raising_hand: Do you want to develop this feature yourself?

- [ ] Yes
- [x] No

**Comment by cortinico** [comment]
> In its current form, the main chucker artifact requires consumers to be on the same versions of the Android/Jetpack libraries that the Chucker UI relies on (and pulling in those transitive dependencies, when not on the same versions, can cause unintended side effects).

Agree. I think the underlying problem is more that we import libraries which ends up affecting the Debug Classpath (e.g. Material Design Components). If we would have no dependencies on external libraries we depend on, we won't have this problem.

**Comment by erawhctim** [comment]
@cortinico any interest/movement on solving this? Would you consider an external PR to address this?

I'd be interested in tackling it, but it's unclear to me what the best path forward is: do we modularize things and let consumers depend on a UI-less artifact (and ship their own UI), do we try and remove MDC as a dependency in the current module structure, or something else entirely? Would appreciate any thoughts you have!

**Comment by cortinico** [comment]
> any interest/movement on solving this? Would you consider an external PR to address this?

I think it would be great to have this solved, but I currently don't have the time to invest on this. I'm more than happy to receive a PR that implements this.

Perhaps first would be ideal to have a diagram of how you envision the codebase to be modularized

**Comment by erawhctim** [comment]
@cortinico would you be interested accepting PRs to migrate to Compose UI & the compose Material artifacts?

I'm thinking through my original problem again (MDC version conflict), and considering a migration from Material Design Components -> Compose UI as an alternative way to solve the problem. If there's a goal to move Chucker over to Compose UI eventually _anyway_, perhaps that could solve two problems at once.
